apps/emqx_resource/src/emqx_resource.erl

```diff
@@ -227,9 +227,9 @@ restart(InstId, Opts) ->
 stop(InstId) ->
     call_instance(InstId, {stop, InstId}).
 
--spec health_check(instance_id()) -> ok | {error, Reason :: term()}.
-health_check(InstId) ->
-    call_instance(InstId, {health_check, InstId}).
```

Comment:
Do not remove `health_check/1`, delegate it to `health_check/2` using `infinity` as the second parameter.